kolibri/plugins/learn/assets/src/store.js

```diff
@@ -1,42 +1,39 @@
 
+const constants = require('./constants');
 const Vuex = require('vuex');
 
 function initialState() {
   return {
-    topics: [],
-    contents: [],
-    full: {
-      title: '',
-      description: '',
-      kind: '',
-      files: [],
-      pk: 0,
-      available: false,
-      extra_fields: '{}',
-      content_id: '',
-    },
+    rootTopicId: global.root_node_pk,
+    pageName: constants.PageNames.EXPLORE_ROOT,
+    pageMode: constants.PageModes.EXPLORE,
+    pageState: {},
+    error: '',
+    loading: true,
   };
 }
 
 const mutations = {
-  /**
-   * Mutation to set all the attributes of a content item retrieved from the server onto
-   *  the 'full' key on the Store state.
-   * @param {Object} state - A Vuex state object.
-   * @param {Object} attributes - The object of key value pairs for the content node.
-   * @constructor
-   */
-  SET_FULL_CONTENT(state, attributes) {
-    Object.assign(state.full, attributes);
+  SET_PAGE_NAME(state, name) {
+    state.pageName = name;
   },
-  SET_CHANNEL(state, channelId) {
-    state.channelId = channelId;
+  SET_PAGE_MODE(state, mode) {
+    state.pageMode = mode;
   },
-  SET_TOPICS(state, topics) {
-    state.topics = topics;
+  SET_LOADING(state) {
+    state.pageState = {};
+    state.error = '';
+    state.loading = true;
   },
-  SET_CONTENTS(state, contents) {
-    state.contents = contents;
+  SET_PAGE_STATE(state, pageState) {
+    state.pageState = pageState;
```

Comment:
Still exploring that - haven't confirmed that this is completely workable. However I'm optimistic based on [the docs](http://vuex.vuejs.org/en/mutations.html#mutations-follow-vues-reactivity-rules) which say it's okay to replace an object with a new one.